ADR/0033-apps-continuous-chaos-testing.md

```diff
@@ -0,0 +1,47 @@
+# 18. Continuous Chaos Testing of Apps in AppStudio
+
+Date: 2024-03-05
+
+## Status
+
+In consideration
+
+## Context
+
+There are a couple of false assumptions that users might have when operating and running their applications in distributed systems: 
+
+Software is Resilient, The network is reliable. There is zero latency. Bandwidth is infinite. The network is secure. Topology never changes. The network is homogeneous. Consistent resource usage with no spikes. All shared resources are available from all places. Various assumptions lead to a number of outages in production environments. The services suffered from poor performance or were inaccessible to the customers, leading to missing Service Level Agreement (SLA) uptime promises, revenue loss, and a degradation in the perceived reliability of said services. How can we best avoid this from happening? This is where Chaos testing can add value.
+
+Failures in production are costly. To help mitigate risk to service health, consider the following strategies and approaches to service testing:
+
+Be proactive vs reactive. We have different types of test suites in place - unit, integration and end-to-end - that help expose bugs in code in a controlled environment. Through implementation of a chaos engineering strategy, we can discover potential causes of service degradation. We need to understand the systems’ behavior under unpredictable conditions in order to find the areas to harden, and use performance data points to size the clusters to handle failures in order to keep downtime to a minimum.
+
+### Glossary
+
+- krkn: Chaos testing framework: <https://github.com/krkn-chaos/krkn>
+
+## Decision
+
+There are two approaches to chaos testing in the CI/CD pipeline.
```

Comment:
This is the first sentence of the decision section. To make it super clear, can you reword it as a decision? For example, is this the decision you're seeking agreement on?

> We are going to adopt chaos testing as a part of the [e2e-tests](https://github.com/redhat-appstudio/e2e-tests/) suite, and execute those tests to gate merges to main in the [infra-deployments](https://github.com/redhat-appstudio/infra-deployments) repo.

Or, is the decision something else, like:

> Because we think we will learn something from krkn, we are going to run some experiments with krkn and generate a report on where the weak areas in konflux are that need to be hardened. Depending on the result of those tests, we may submit a followup ADR recommending that we integrate krkn in our e2e-tests to gate merges to main in infra-deployments.

Or, perhaps I misunderstand, and the ADR is a recommendation to adopt krkn _as a part of the product_, perhaps in [integration-service](https://github.com/redhat-appstudio/integration-service)?  Are you arguing here that the decision should be something like:

> We recommend krkn as a testing tool for users of Konflux to use in their IntegrationTestScenarios, so that our users can reap the benefits of chaos testing on their applications - orchestrated by the integration-service in the konflux platform.

At the moment, this ADR reads more like an informative blog post (and, it is good and informative) but it doesn't feel like a decision to be recorded.

Reply:
Hello @ralphbean I have refactored the ADR based on your review comments. I would appreciate it if you could review the latest changes and provide your feedback.

Reply:
Much clearer now - thank you!